### PR TITLE
tikv_utils, *: merge `after_start_wrapper` and `before_stop_wrapper`.

### LINKED ISSUE #15014: tikv-util: the result of wrappers will be inconsistent once the user partially register the hooks.
## Development Task
Once the user call:
```
tokio::runtime::Builder::new_multi_thread()
  .after_start_wrapper(|| {...})
  .build()
```
This will result in some system hooks at thread start being called, but not at exit.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #15014 

What's Changed:
Merged `after_start_wrapper` and `before_stop_wrapper`.

```commit-message 
Replace `after_start_wrapper` and `before_stop_wrapper` with `with_sys_and_custom_hooks`.
Replace no-op(how strange!) `after_start_wrapper` and `before_stop_wrapper` with `with_sys_hooks`.
```

### Check List 

Tests 

- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/backup-stream/src/endpoint.rs [hunk 1]
```diff
@@ -1026,13 +1026,11 @@ fn create_tokio_runtime(thread_count: usize, thread_name: &str) -> TokioResult<R
...
         .worker_threads(thread_count)
         .enable_io()
-        .enable_time()
-        .after_start_wrapper(|| {
-            tikv_alloc::add_thread_memory_accessor();
-        })
-        .before_stop_wrapper(|| {
-            tikv_alloc::remove_thread_memory_accessor();
-        })
+        .enable_time().with_sys_and_custom_hooks(|| {
```
Comment (YuJuncen) [inline components/backup-stream/src/endpoint.rs:1029]: It's pretty unexpected that rustfmt won't complain this... 🙃

**Review by tabokie — APPROVED** [review]
Need to update clippy.toml, rest LG.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by YuJuncen** [comment]
@tabokie PTAL

**Pull request merged**